Az árverés nevének beállítása backendből

remenyi opened the pull request:
A frontenden egy auction service-t hoztam létre, ez a komponens éri el a backendet. A backenden a mongoose modullal érem el a MongoDb-t.

A NodeJs Express alapértelmezetten nem teljesít egyéb domainekről érkező kéréseket, de ezt megoldottam a cors nevű csomaggal.

Mivel még nem találtuk ki, hogy hogyan és honnan töltjük fel az adatbázist, ezért a NodeJs Express-t elindító kódba tettem az adatbázist egyetlen aukció objektummal feltöltő kódot. Sajnos nem jöttem rá, hogy hogyan lehetne kitörölni az előző indításokból az adatbázisban maradt objektumokat, ezért a NodeJs Express újraindításakor mindig keletkezik egy új objektum. Egyébként MongoDB Compass-ban meg lehet nézni, hogy tényleg létrejött az objektum (és a NodeJs Express újraindítása után azt is lehet látni - hibásan -, hogy több példányban van jelen az objektum), és a name kulcshoz tartozó mezőt megváltoztatva a változás megjelenik a frontenden is.

remenyi committed:
Set auction name from backend

On the frontend the auction service fetches the name for the auction from '/api/auction'. On the backend the auction name is fetched from 'auctions-dev' MongoDb database. Additionally, 'localhost:4200' was enabled on the backend.

zoltan31 commented on api/src/index.ts:
```diff
@@ -3,10 +3,46 @@ import * as express from "express";
 const port = 5000;
 
+// enabling 'http://localhost:4200' to use the api
+var cors = require('cors');
```
A require() függvény helyett próbáljuk meg az import-ot használni. Ugyanígy a 'var' helyett a 'const', illetve 'let' kulcsszavakat. Ez esetben ez átírható erre:
`import cors from "cors";`

zoltan31 commented on api/src/index.ts:
```diff
@@ -3,10 +3,46 @@ import * as express from "express";
+app.use(cors(corsOptions));
+
+// connect to DB
+const mongoose = require("mongoose");
```
Ugyanaz mint előbb

zoltan31 commented on api/src/index.ts:
```diff
@@ -3,10 +3,46 @@ import * as express from "express";
+db.on('error', (error) => console.error(error));
+db.once('open', () => console.log('Connected to Database'));
+
+const Auction = require('./models/auction');
```
Ugyanaz mint előbb

zoltan31 commented on api/src/index.ts:
```diff
@@ -3,10 +3,46 @@ import * as express from "express";
+// insert new object into DB
+// I don't know how to delete objects from previous sessions 😅
+// But it works for now
+const first_auction = new Auction ({name: "Auction Name From MongoDB🔥"});
```
Változónevek legyenek camelCase formájúak a snake_case helyett.

zoltan31 commented on api/src/models/auction.ts:
```diff
@@ -0,0 +1,8 @@
+const mongoose = require("mongoose")
```
Ugyanaz mint előbb.

zoltan31 commented on api/src/models/auction.ts:
```diff
@@ -0,0 +1,8 @@
+    name: String
+})
+
+module.exports = mongoose.model('Auction', auctionSchema)
```
Itt pedig a default exportot használjuk inkább:
`export default mongoose.model('Auction', auctionSchema);`

zoltan31 commented on api/src/index.ts:
```diff
@@ -3,10 +3,46 @@ import * as express from "express";
+const Auction = require('./models/auction');
+
+// insert new object into DB
+// I don't know how to delete objects from previous sessions 😅
```
Próbálj meg olyat, hogy nézd meg mentés előtt, hogy van-e ilyen nevű Auction. Ha van, akkor töröld (vagy ne töröld és ne is tegyél be az adatbázisba). Nem ellenőriztem le, de valami ilyesmi lehetne:
```
if (!Auction.exists({name: "Auction Name From MongoDB🔥"})) {
  first_auction.save();
}
```

zoltan31 reviewed (changes requested):
Nem futtattam le, van egy-két konvenciótól való eltérés, viszont jónak tűnik.

remenyi committed:
Change 'requires' to 'imports...' and eliminate 'vars'

Changed package import mechanism from 'required...' to 'import...'.
Some imported packages - mongoose and cors - had to be changed for this to work.
Changed var(s) to const(s) or let(s).

remenyi committed:
Ensure only one auction object is in the database

Only inserting an auction object into the database, if there are none in the database
